apps/meteor/server/routes/health.ts

```diff
@@ -1,24 +1,120 @@
 
-import { isRunningMs } from '../lib/isRunningMs';
 import { SystemLogger } from '../lib/logger/system';
 
-WebApp.rawConnectHandlers.use('/health', async (_req, res) => {
+/**
+ * A shared handler for liveness probes to reduce code duplication.
+ * Sets common headers and sends a standard 200 OK response.
+ * @param res The HTTP response object.
+ */
+const sendLivenessResponse = (res: ServerResponse) => {
+	// Set headers to prevent any caching of the health check response.
 	res.setHeader('Cache-Control', 'no-cache, no-store, must-revalidate');
 	res.setHeader('Pragma', 'no-cache');
-	res.setHeader('Expires', 0);
-	res.setHeader('Content-Type', 'text/plain');
-
-	if (!isRunningMs()) {
-		const { isLastDocDelayed } = await import('../startup/watchDb');
-
-		if (isLastDocDelayed()) {
-			SystemLogger.error('System not healthy due to no real time data received recently');
-			res.writeHead(500);
-			res.end('not healthy');
-			return;
-		}
+	res.setHeader('Expires', '0');
+	res.setHeader('Content-Type', 'application/json');
+
+	res.writeHead(200);
+	res.end(JSON.stringify({ status: 'ok' }));
+};
+
+/**
+ * Liveness Probe (`/livez`)
+ * Responds with 200 OK if the process is running.
+ * Used by orchestrators (Kubernetes) to decide whether to restart the container.
+ */
+WebApp.rawConnectHandlers.use('/livez', (_req: IncomingMessage, res: ServerResponse) => {
+	sendLivenessResponse(res);
+});
+
+/**
+ * DEPRECATED Liveness Probe (`/health`)
+ * Maintained for backward compatibility. Behaves like the /livez probe.
+ * @deprecated Update infrastructure to use /livez and /readyz.
+ */
+WebApp.rawConnectHandlers.use('/health', (_req: IncomingMessage, res: ServerResponse) => {
+	SystemLogger.info('Deprecated /health endpoint was called. Please update to /livez or /readyz.');
+	sendLivenessResponse(res);
+});
+
+/**
+ * Readiness Probe Configuration
+ * Defines tunable thresholds for readiness checks.
+ */
+
+// Initialize and enable the event loop delay histogram for accurate monitoring.
+const eventLoopHistogram = monitorEventLoopDelay();
+eventLoopHistogram.enable();
+
+// Configurable thresholds for readiness checks.
+const READINESS_THRESHOLDS = {
+	EVENT_LOOP_LAG_MS: Number(process.env.EVENT_LOOP_LAG_MS) || 70,
+	HEAP_USAGE_PERCENT: Number(process.env.HEAP_USAGE_PERCENT) || 0.85, // 85%
+};
+
+/**
+ * Checks event loop lag, a key indicator of Node.js saturation.
+ * @returns The status and p99 lag.
+ */
+function checkEventLoopLag() {
+	// Get the 99th percentile value from the histogram (in nanoseconds).
+	const lagInNs = eventLoopHistogram.percentile(99);
+	const lagInMs = lagInNs / 1_000_000; // Convert to milliseconds
+
+	// Reset the histogram to measure the next interval cleanly.
+	eventLoopHistogram.reset();
+
+	if (lagInMs > READINESS_THRESHOLDS.EVENT_LOOP_LAG_MS) {
+		return { status: 'degraded', lag: `${lagInMs.toFixed(2)}ms (p99)` };
+	}
+	return { status: 'ok', lag: `${lagInMs.toFixed(2)}ms (p99)` };
```

Comment:
same here

Reply:
Nothing is really going to be consuming this.  This is more for our ability to look if we ever want to.